### DISCUSSION TITLE
Add relationship() support to TextColumn

### DISCUSSION THREAD

**Discussion by somzoli** [filamentphp/filament#10309] [Ideas]
I have two models:

- Rule
- Interface

Relations:
Rule:
`public function IInterfaces() { return $this->belongsTo(IInterface::class, 'i_interface_id'); }`
Interface:
`public function Rules() { return $this->HasMany(Rule::class); }`

But the interface model is usable as input interface and out interface. For this i have to add an extra foreign key to Rule table:

```
//IN
$table->foreignIdFor(Models\IInterface::class)->nullable()->cascadeOnDelete();
 //OUT
$table->foreignIdFor(Models\IInterface::class, 'out_i_interface_id')->nullable()->cascadeOnDelete();
```
In Rule form, i can use this, which is working fine:
`Forms\Components\Select::make('out_i_interface_id') ->relationship('IInterfaces', 'name') ->label('OUT interface'),`

But in Columns on Rule list, the "OUT" column can not use, because i cant define the relation that i want to use
`Tables\Columns\TextColumn::make('out_i_interface_id') ->label('Out Interface') ->searchable() ->sortable(),`
Describe the solution you'd like
Maybe can port relationship() from Select?

Can you help with this?
Thanks!

BR,
Z

**Comment by somzoli** [comment]
Hi,

I was able to solve this with add extra relations to Rule model:
`public function OutIInterfaces()
    {
        return $this->belongsTo(IInterface::class, 'out_i_interface_id');
    }`

**Comment by sax1987** [comment]
I tried your process, but I got a data type problem, can you help me?

**Reply by somzoli** [reply]
Hi,

describe your issue, i can try to help.

**Reply by sax1987** [reply]
Thank you very much, I have done everything you suggested, but it shows me this error. as you see in the images

**Reply by sax1987** [reply]
![image](https://github.com/filamentphp/filament/assets/164684917/fb1b93ae-ff8e-4174-88ab-63cc1ec1abb2)
![image](https://github.com/filamentphp/filament/assets/164684917/8cdeba4d-66a2-454f-b2f2-1a2602b7d179)
![image](https://github.com/filamentphp/filament/assets/164684917/b6d13d78-4c9f-4d61-908c-f1330d6b498b)
![image](https://github.com/filamentphp/filament/assets/164684917/894b3a5f-f537-4613-aaa0-26252eaeaf99)
![image](https://github.com/filamentphp/filament/assets/164684917/083eb98c-c9d0-4862-a4ed-db3cf48c1bfd)

**Reply by sax1987** [reply]
![image](https://github.com/filamentphp/filament/assets/164684917/2a46df57-8aa7-449d-ba90-500635fc28e1)

**Reply by sax1987** [reply]
I hope the images help you understand the problem a little.

**Reply by sax1987** [reply]
my person table would be your interface table

**Reply by somzoli** [reply]
Hi,

I think you have a typo issue. 
When you define the relation in ->relationship() method, you have to give a table name instead of model name. 

Test with like this:

```
Forms\Components\Select:: make ('persona id')
// -›options(Persona::all()->pluck( 'nombre_apellido', 'id'))
-›relationship ( 'personas', 'nombre_apellido')
->label (' Comprador' )
-›searchable()
->preload()
-›required(),
Forms\Components\Select:: make ('v persona id')
//-›options (Persona: :all() ->pluck('nombre_apellido', 'id'))
-›relationship ('personas', 'nombre_apellido' )
->label ('Vendedor')
-›searchable()
->preload ()
-yrequired（）.
Forms \Components\Select: make('r_persona_id' )
// -›options (Persona::all(->pluck( 'nombre_apellido', 'id'))
-›relationship( 'personas', 'nombre _apellido')
->label ('Responsable' )
-›searchable ()
->preload()
-›required(),

```

**Reply by somzoli** [reply]
Other issue can be:

if you use v_persona_id in textcolumn, u should use VPersonas function name in related model.

filament will calculate lile this:
VPersonas-> v_persona_id

```
public function VPersonas (): BelongsTo
return $this-›belongsTo(Persona: :class, 'v_persona_id');
```

**Reply by sax1987** [reply]
I already found the solution my friend thanks for the help
